handler.go

```diff
@@ -291,17 +291,34 @@ func (h BinaryHTTPAppHandler) Handle(binaryRequest []byte, metrics Metrics) ([]b
 	return binaryRespEnc, r
 }
 
+// TargetRewrite represents a rewritten target request.
+type TargetRewrite struct {
+	Scheme string
+	Host   string
+}
+
 // HttpRequestHandler handles HTTP requests to produce responses.
 type HttpRequestHandler interface {
 	// Handle takes a http.Request and resolves it to produce a http.Response.
 	Handle(req *http.Request, metrics Metrics) (*http.Response, error)
 }
 
+// HttpClientRequestHandler represents a HttpRequestHandler that handles requests by sending them
+// with an http.Client
+type HttpClientRequestHandler struct {
+	client *http.Client
+}
+
+func (h HttpClientRequestHandler) Handle(req *http.Request, metrics Metrics) (*http.Response, error) {
+	return h.client.Do(req)
+}
```

Comment:
HttpClientRequestHandler is a thin wrapper around an `*http.Client`. Why do we need it? IT doesn't seem to be necessary in `handler.go`: is it used by the new unit tests?

Reply:
Correct. `MockHTTPRequestHandler`, defined in `gateway_test.go`, also conforms to this interface but it returns a deterministic response to the test harness, as opposed to this implementation which uses its HTTP client to send `req` along to the target.